### PR TITLE
[temp] Replace \term with \defn or \placeholder

### PR DESCRIPTION
**Pull request by jensmaurer**
Thereby also improving the index.

Partially addresses #329.

### CODE REVIEW

File: source/templates.tex [hunk 1]
```diff
@@ -3556,11 +3561,11 @@
 \rSec2[temp.dep]{Dependent names}
 
 \pnum
-\indextext{name!dependent}%
+\indextext{dependent name|see{name!dependent}}%
```
Comment (Eelis) [inline source/templates.tex]: Should this be `see{name, dependent}`?

Comment (jensmaurer) [inline source/templates.tex]: Yes, good catch.

**Comment by jensmaurer** [comment]
Updated and rebased.

File: source/templates.tex [hunk 2]
```diff
@@ -3704,7 +3708,8 @@
 
 \pnum
 A name refers to the
-\term{current instantiation}
+\indextext{current instantiation|see{instantiation, current}}
+\defnx{current instantiation}{instantiation!current}
```
Comment (zygoloid) [inline source/templates.tex]: I don't think grouping this under "instantiation" is appropriate; the "current instantiation" is a notion separate from instantiation itself (in fact, it's most relevant when checking a template definition prior to performing any instantiation).

Comment (jensmaurer) [inline source/templates.tex]: You're right.

**Comment by jensmaurer** [comment]
Fixed as requested.

File: source/templates.tex [hunk 3]
```diff
@@ -3808,9 +3812,9 @@
 \end{note}
 
 \pnum
-\indextext{instantiation!member of the current}%
+\indextext{member of the current instantiation|see{instantiation, member of the current}}%
 A name is a
-\term{member of the current instantiation}
+\defnx{member of the current instantiation}{instantiation!member of the current}
```
Comment (zygoloid) [inline source/templates.tex]: This should be `{current instantiation!member of}` or similar.

File: source/templates.tex [hunk 4]
```diff
@@ -3863,14 +3867,14 @@
 \end{codeblock}
 \end{example}
 
-\indextext{instantiation!dependent member of the current}%
-A name is a \term{dependent member of the current instantiation} if it is a
+\indextext{dependent member of the current instantiation|see{instantiation, dependent member of the current}}%
+A name is a \defnx{dependent member of the current instantiation}{instantiation!dependent member of the current} if it is a
```
Comment (zygoloid) [inline source/templates.tex]: Likewise.

**Comment by jensmaurer** [comment]
Fixed the two "member of the current instantiation" issues. The index looks sane now (in this regard).

**Pull request merged**